kinto/core/authentication.py

```diff
@@ -37,6 +44,93 @@ def unauthenticated_userid(self, request):
             return userid
 
 
+@implementer(IAuthenticationPolicy)
+class OpenIDConnectPolicy(base_auth.CallbackAuthenticationPolicy):
+    def __init__(self, realm='Realm'):
+        self.realm = realm
+        self._cache = None
+
+        self._jwt_keys = None
+
+    def unauthenticated_userid(self, request):
+        """Return the FxA userid or ``None`` if token could not be verified.
+        """
+        authorization = request.headers.get('Authorization', '')
+        try:
+            authmeth, payload = authorization.split(' ', 1)
+        except ValueError:
+            return None
+
+        if authmeth.lower() != 'bearer+oidc':
+            return None
+
+        try:
+            # Bearer+OIDC id_token=jwt, access_token=bearer
+            parts = payload.split(',')
+            credentials = {k: v for k, v in [p.strip().split("=") for p in parts]}
+            id_token = credentials["id_token"]
+            access_token = credentials["access_token"]
+        except (ValueError, KeyError):
+            # Bearer+OIDC jwt
+            id_token = payload
+            access_token = None
+
+        return self._verify_token(request, id_token, access_token)
+
+    def forget(self, request):
+        """A no-op. Credentials are sent on every request.
+        Return WWW-Authenticate Realm header for Bearer token.
+        """
+        return [('WWW-Authenticate', 'Bearer+OIDC realm="%s"' % self.realm)]
+
+    def _verify_token(self, request, id_token, access_token):
+        # Verify signature
+        audience = request.registry.settings["oidc.audience"]
+        issuer = request.registry.settings["oidc.issuer_url"]
+
+        if self._jwt_keys is None:
+            resp = requests.get(issuer.rstrip("/") + "/.well-known/openid-configuration")
+            jwks_uri = resp.json()["jwks_uri"]
+            resp = requests.get(jwks_uri)
+            self._jwt_keys = resp.json()["keys"]
+
+        try:
+            unverified_header = jwt.get_unverified_header(id_token)
+        except jwt.JWTError:
+            logger.debug("Invalid header. Use an RS256 signed JWT Access Token")
+            return None
+
+        if unverified_header["alg"] != "RS256":
+            logger.debug("Invalid header. Use an RS256 signed JWT Access Token")
+            return None
+
+        rsa_key = {}
+        for key in self._jwt_keys:
+            if key["kid"] == unverified_header["kid"]:
+                rsa_key = utils.dict_subset(key, ("kty", "kid", "use", "n", "e"))
+                break
+        if not rsa_key:
+            logger.debug("Unable to find appropriate key")
+            return None
+        try:
+            options = None
+            if access_token is None:
+                options = {"verify_at_hash": False}
+
+            payload = jwt.decode(id_token, rsa_key, algorithms=["RS256"],
```

Comment:
I've logged the code and verified that the error `at_hash claim missing from token` is being raised from this function.
I cloned (https://github.com/leplatrem/python-jose/tree/75-at_hash-optional)[leplatrem python-jose] implementation that is supposed to address the issue in order to test, but I don't know if the changes on code is being reflected on kinto source

Reply:
Sorry @cfguimaraes for the delay...

Basically you should be able to apply the fork with:
```
$ cd kinto/
$ source .venv/bin/activate
$ pip install -e <path-to-python-jose-fork>
```

But according to the code above if you provide a JWT but no access token, then this hash is not verified.

The status of this branch is still very rough, I want to land it ASAP, we'll need it in our production stacks at Mozilla :)